onnxruntime/core/framework/debug_node_inputs_outputs_utils.cc

```diff
@@ -117,17 +128,157 @@ void DumpTensorToFile(const Tensor& tensor, const std::string& tensor_name, cons
   ORT_THROW_IF_ERROR(Env::Default().FileClose(output_fd));
 }
 
+bool TensorExistsInSqlDb(const TensorMetadata& tensor_metadata) {
+
+  static std::unique_ptr<sqlite3_stmt, decltype(&sqlite3_finalize)> stmt_uptr( [](){
```

Comment:
>stmt_uptr [](http://example.com/codeflow?start=68&length=9)

are there thread-safety concerns with using a static connection/statement? perhaps this could be part of a context object?

Reply:
This is unlikely to be thread-safe. For training we always use single thread sequential executor. I will think what might be a solution.

Reply:
Uses thread_local for storage and so should be thread-safe. For now the utility is not injected into parallel executor and so won't cause any user issues. It could be better tested when that use case is considered.